packages/jsxstyle/jsxstyle.js

```diff
@@ -27,7 +27,12 @@ function factory(displayName, defaultProps, tagName) {
     }
 
     render() {
-      const { props, style, children } = this.props;
+      const { props = {}, style, children, domRef } = this.props;
+
+      if (typeof this.component === 'string' && domRef) {
```

Comment:
Any particular reason behind only adding the `ref` prop to DOM components? Seems to me like it should always be passed, but I might be misunderstanding something. If someone specifies a React class as `component` and specifies a `domRef` as well, things might get a bit confusing when `domRef` doesn’t do anything.

Maybe the prop should be called something that’s not tied to “DOM”? `innerRef`?

Reply:
I had originally passed `domRef` through as a prop if it was specified and the component wasn't a string.  Looks like that got lost while I was testing, but I'll go put it back.  Thanks for the catch.

I'm skeptical of the value of an `innerRef` prop if it doesn't refer to the DOM element.  (I'm also skeptical of the value of a `ref` that refers to a component, but that's a separate conversation.)  

There are times when it's really useful to have a reference to a vanilla DOM element - for instance, to listen for events that React doesn't expose props for.  Thus, it's valuable to be able to get a reference to the DOM element that a component resolves to, no matter how many layers deep it the definition is.

If we had an `innerRef`, you'd have to know about the inner workings of `component` to know what it would actually resolve to.  I'm not sure that's useful, and it's not well encapsulated.

For example, the same `domRef` function works in both of these cases, regardless of what type `component` is: 

```typescript
function listenToPointerEvents(element) { … }

<Row
  domRef = { listenToPointerEvents }
/>
<Row
  domRef = { listenToPointerEvents }
  component = { EventuallyResolvesToADOMNode }
/>
```

If we instead used `innerRef`:

```typescript
<Row
  innerRef = { function (maybeDOMElementOrMaybeComponentInstance) {…} }
  component = { EventuallyResolvesToADOMNode }
/>
```

You'd have to know the shape of `EventuallyResolvesToADOMNode`, and know which node it passes `innerRef` to.  That doesn't seem useful.

Reply:
Pushed a new version that passes through `domRef` if `component` isn't a string.